src/server/routes/v1/index.ts

```diff
@@ -1,110 +1,1213 @@
+
+  // main visitor routes
+  fastify.post("/register", async (request: FastifyRequest, reply: FastifyReply<ServerResponse>) => {
+    // bandsy and oauth acc type route
+
+    const { accountType }: IRegisterRouteBody = request.body;
+
+    if (accountType === UserAccountType.BANDSY) {
+      const { email, password }: IRegisterRouteBody = request.body;
+
+      if (email == null || password == null) {
+        reply.code(400);
+
+        return {
+          error: "email or password not specified",
+        };
+      }
+
+      try {
+        const { salt, passwordHash } = await generateSaltedPassword(password);
+
+        // TODO: maybe do something so createdAt and updatedAt get set automatically
+        // TODO: also maybe make it so those fields cant be tampered with?
+        const { uuid } = await userService.create({
+          accountType,
+
+          email,
+          verified: false,
+
+          salt,
+          passwordHash,
+
+          mfaEnabled: false,
+
+          createdAt: new Date(),
+          updatedAt: new Date(),
+        });
+
+        // yeet old verification codes first (scenario: someone makes an acc, deletes in, makes a new one in an instant,
+        // now they have 2 verification codes!)
+        await verificationService.deleteMany({
+          userEmail: email,
+
+          type: IVerificationType.VERIFICATION,
+        });
+
+        const verificationCode = await generateRandomToken(24);
+
+        await verificationService.create({
+          userUuid: uuid,
+          userEmail: email,
+
+          code: verificationCode,
+          type: IVerificationType.VERIFICATION,
+          // TODO: set this to a constant and make it longer than 5 mins lmao
+          validUntil: new Date(new Date().getTime() + (1000 * 60 * 5)),
+
+          createdAt: new Date(),
+          updatedAt: new Date(),
+        });
+
+        // TODO: put an actual username there?
+        const verificationEmail = new VerificationEmail(transportOptions);
+        await verificationEmail.send(email, {
+          username: "cunty mcjim",
+          verificationCode,
+        });
+      } catch (error) {
+        reply.code(500);
+
+        return {
+          error: `error processing your request: ${process.env.NODE_ENV === "dev" ? error : "()"}`,
+        };
+      }
+
+      reply.code(204);
+
+      return null;
+    }
+
+    if (accountType === UserAccountType.OAUTH) {
+      const { oauthService, accessToken }: IRegisterRouteBody = request.body;
+
+      if (oauthService == null || accessToken == null) {
+        reply.code(400);
+
+        return {
+          error: "oauth service type or access token not specified",
+        };
+      }
+
+      // check this as exchangeToken expects an enum value, which cant be checked at compile
+      // time cos user input and such, passing an incorrect value wont break anything, it just
+      // wont throw a pretty looking error like the one below ^^
```

Comment:
either you're retarded or missing a mouth

Reply:
![image](https://user-images.githubusercontent.com/18635887/79694214-ddf02780-8266-11ea-8f3e-047e68df0d1d.png)